crates/solver/src/s3_instance_upload.rs

```diff
@@ -40,14 +42,25 @@ impl S3InstanceUploader {
     /// Upload the bytes (expected to represent a json encoded solver instance) to the configured S3
     /// bucket.
     ///
-    /// The final filename is the configured prefix followed by `{current_date}/{auction_id}`.
+    /// The final filename is the configured prefix followed by `{auction_id}.json.gzip`.
     pub async fn upload_instance(&self, auction: AuctionId, value: Vec<u8>) -> Result<()> {
+        let (key, bytes) = self.prepare_instance(auction, &value)?;
+        self.upload(key, bytes).await
+    }
+
+    /// Returns the S3 key and body after gzipping.
+    fn prepare_instance(&self, auction: AuctionId, json: &[u8]) -> Result<(String, Vec<u8>)> {
         let key = self.filename(auction);
-        self.upload(key, value).await
+        // This function isn't async because we already have the body in memory and compressing is
+        // fast.
+        let mut encoder = GzEncoder::new(json, Compression::best());
+        let mut encoded: Vec<u8> = Vec::with_capacity(json.len());
+        encoder.read_to_end(&mut encoded).context("gzip encoding")?;
+        Ok((key, encoded))
     }
 
     fn filename(&self, auction: AuctionId) -> String {
-        format!("{}{auction}.json", self.filename_prefix)
+        format!("{}{auction}.json.gzip", self.filename_prefix)
```

Comment:
The auction ids could collide across staging/prod. Do we have separate S3 buckets or some other strategy to deal with that? I guess this could also be handled with different `filename_prefix`es.

Reply:
Also do you think we should compress all the files we already have as a follow up task to have a consistent file format?

Reply:
> The auction ids could collide across staging/prod. 

The intention is to have the prefix handle conflicts

> Also do you think we should compress all the files we already have as a follow up task to have a consistent file format?

No. My plan is to enable the new code (it isn't enabled yet) and leave the old bucket as is until it expires and we delete it.